### PR TITLE
Log tracing

### PR DESCRIPTION
**Pull request by swalchemist**
Log tracing - this changes the UAA log format used in a bosh uaa deployment to include a trace ID and span ID, and adjusts the tests to match.

This should be merged after https://github.com/cloudfoundry/uaa/pull/2446 in order for the tests to pass. See that PR for details about the feature.

[#185000950]

### CODE REVIEW

**Comment by bruce-ricard** [comment]
Do we want to have a separate UAA repo PR for the source code changes, rather than reviewing them from here through the submodule? I find it a bit awkward. 

I was the one who bumped the submodule in the uaa-release feature branch, though it was for ease of developer use while working on it, not necessarily for the PR.

edit: I think that we should make a separate commit: it's essentially impossible to comment on the UAA changes.

File: jobs/uaa/templates/config/log4j2.properties.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@ case logging_format_timestamp_value
 end
 %>
 property.log_directory = /var/vcap/sys/log/uaa
-property.log_pattern=[<%= timestamp_format %>] uaa%X{context} - %pid [%t] .... %5p --- %c{1}: %replace{%m}{(?<=password=|client_secret=)([^&]*)}{<redacted>}%n
+property.log_pattern=[<%= timestamp_format %>] uaa%X{context} - %pid [%t] - [%X{traceId},%X{spanId}] .... %5p --- %c{1}: %replace{%m}{(?<=password=|client_secret=)([^&]*)}{<redacted>}%n
```
Comment (bruce-ricard) [inline jobs/uaa/templates/config/log4j2.properties.erb:18]: This is technically a breaking change. Do we want to make it configurable in the spec? @strehle any thoughts?

**Review by bruce-ricard — CHANGES_REQUESTED** [review]

File: jobs/uaa/templates/config/log4j2.properties.erb (hunk 1 above)
Comment (strehle) [reply jobs/uaa/templates/config/log4j2.properties.erb:18]: in what sense breaking ? but if you have doubts then yes please add a config for it

**Comment by swalchemist** [comment]
> Do we want to have a separate UAA repo PR for the source code changes, rather than reviewing them from here through the submodule? I find it a bit awkward.

There is a separate PR for UAA - https://github.com/cloudfoundry/uaa/pull/2446. Committing the submodule change here was probably not helpful, and we shouldn't review specific uaa changes in a uaa-release PR anyway.

File: jobs/uaa/templates/config/log4j2.properties.erb (hunk 1 above)
Comment (bruce-ricard) [reply jobs/uaa/templates/config/log4j2.properties.erb:18]: It could possibly break a user's log parser. Although it would be pretty easy to fix.

We don't really want to make it configurable, we think it's OK.

Comment (swalchemist) [reply jobs/uaa/templates/config/log4j2.properties.erb:18]: Anything that changes the log format could break existing code that parses the logs. We asked around internally and no one was concerned about it as long as we mentioned the change in the release notes - it seems that our customers are used to updating their log parsing when they install an update. 

So I'm inclined not to make this configurable unless someone has a solid need for it.

**Comment by bruce-ricard** [comment]
I removed the UAA change from this PR. We now need to merge the UAA one before we merge this one.

File: scripts/create-dev-release.sh [hunk 2]
```diff
@@ -3,6 +3,9 @@
 set -euo pipefail
 
 export UAA_VERSION=0.0.0
+JAVA_HOME=$(/usr/libexec/java_home -v 11)
```
Comment (hsinn0) [inline scripts/create-dev-release.sh]: Why this change? The script should not be tied with specific Java version.

Comment (hsinn0) [reply scripts/create-dev-release.sh]: Also, this is likely to work only on MacOS. It should be reverted.

**Review by hsinn0 — CHANGES_REQUESTED** [review]

File: scripts/create-dev-release.sh (hunk 2 above)
Comment (Tallicia) [reply scripts/create-dev-release.sh]: @swalchemist Can you remove this?

File: jobs/uaa/templates/config/log4j2.properties.erb (hunk 1 above)
Comment (Tallicia) [reply jobs/uaa/templates/config/log4j2.properties.erb:18]: It appears this is resolved at this point?

**Review by Tallicia — COMMENTED** [review]
These commits look extraneous to the PR feature request.

**Review by hsinn0 — APPROVED** [review]

**Comment by swalchemist** [comment]
I have removed a giant mess of unnecessary commits from this branch. I'm not aware of any actual code change that this causes, but I'm not sure that nothing changed.

File: src/acceptance_tests/uaa-release_test.go [hunk 3]
```diff
@@ -140,9 +140,12 @@ var _ = Describe("UaaRelease", func() {
...
 	})
 
-	logLineWithoutTimestampRegex := ` uaa.* - \d+ \[([^]]+)\] .... (DEBUG|\sINFO|\sWARN) --- .+: .+`
+	logLineWithoutTimestampRegex := ` uaa.* - \d+ \[([^]]+)\] - \[[^]]+\] .... (DEBUG|\sINFO|\sWARN) --- .+: .+`
 
-	DescribeTable("UAA log format", func(uaaLogFormat string, optFiles ...string) {
+	DescribeTable("UAA log format", func(
+		uaaLogFormat string,
+		optFiles ...string,
+	) {
```
Comment (hsinn0) [inline src/acceptance_tests/uaa-release_test.go:148]: I'd rather not see this kind of reformatting as part of the PR.

File: src/acceptance_tests/uaa-release_test.go [hunk 4]
```diff
@@ -292,7 +295,10 @@ func scpUaaAuditLog() string {
 	return localUAALogPath
 }
 
-func getFingerPrint(certdata []byte) (string, error) {
+func getFingerPrint(certdata []byte) (
+	string,
+	error,
+) {
```
Comment (hsinn0) [inline src/acceptance_tests/uaa-release_test.go:301]: The same. I'd rather not see this kind of reformatting as part of the PR.

**Review by hsinn0 — APPROVED** [review]

File: src/acceptance_tests/uaa-release_test.go (hunk 3 above)
Comment (swalchemist) [reply src/acceptance_tests/uaa-release_test.go:148]: Hmmm, interesting. For Go, the reformatting is usually automatic and something that's very difficult to suppress. So the code this was changed probably wasn't changed with an IDE.

Comment (Tallicia) [reply src/acceptance_tests/uaa-release_test.go:148]: If it's coming from the IDE, and they aren't intended line changes, they can can be unselected.
